mining/mqueue.go

```diff
@@ -0,0 +1,114 @@
+package mining
+
+import (
+	"bytes"
+	"container/heap"
+	"sort"
+
+	"github.com/filecoin-project/go-filecoin/address"
+	"github.com/filecoin-project/go-filecoin/types"
+)
+
+// MessageQueue is a priority queue of messages from different actors. Messages are ordered
+// by decreasing gas price, subject to the constraint that messages from a single actor are
+// always in increasing nonce order.
+// All messages for a queue are inserted at construction, after which messages may only
+// be popped.
+// Potential improvements include:
+// - deprioritising messages after a gap in nonce value, which can never be mined (see Ethereum)
+// - attempting to pack messages into a fixed gas limit (i.e. 0/1 knapsack subject to nonce ordering)
```

Comment:
I'm unfamiliar with the "knapsack" terminology; what does this refer to?

Reply:
I added a comment: https://en.wikipedia.org/wiki/Knapsack_problem